### PR TITLE
Add Gradle/Maven wrappers to codestarts

### LINKED ISSUE #11125: Create an api to generate Maven/Gradle wrappers accessible from devtools-commons
**Description**
Create a tool to generate Maven/Gradle wrapper accessible from devtools-commons. This is meant to be use by all the tooling: CLI, code.quarkus.io, maven and gradle plugins

**Implementation ideas**
This should not include the binaries, but have a way to download and cache it.

### LINKED ISSUE #11127: Executing the Gradle wrapper from Codestart integration tests is not working in CI
**Describe the bug**
https://github.com/quarkusio/quarkus/blob/master/integration-tests/maven/src/test/java/io/quarkus/maven/it/CreateProjectCodestartMojoIT.java#L86

The exception is that the wrapper can't find the main class, I can't find it again.

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes https://github.com/quarkusio/quarkus/issues/11127

Closes https://github.com/quarkusio/quarkus/issues/11125

### CODE REVIEW

**Comment by ia3andy** [comment]
I added backport because generated gradle wrapper will not work without this fix

File: devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/gradle/codestart.yml [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+  "gradlew": "executable"
+  "gradlew.bat": "executable"
 language:
   base:
     data:
       quarkus:
         plugin:
           id: io.quarkus
       version:
+        gradle: 6.4.1
```
Comment (gastaldi) [inline devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/gradle/codestart.yml:13]: Quarkus use Gradle 6.5.1

Comment (ia3andy) [reply devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/gradle/codestart.yml]: Good ctach!!! Fixed :)

**Review by gastaldi — APPROVED** [review]
LGTM

File: devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/maven/base/pom.tpl.qute.xml [hunk 2]
```diff
@@ -16,8 +16,8 @@
         <quarkus.platform.artifact-id>{quarkus.platform.artifact-id}</quarkus.platform.artifact-id>
         <quarkus.platform.version>{quarkus.platform.version}</quarkus.platform.version>
         <quarkus-plugin.version>{quarkus.plugin.version}</quarkus-plugin.version>
-        <compiler-plugin.version>{maven.version.maven-compiler-plugin}</compiler-plugin.version>
-        <surefire-plugin.version>{maven.version.maven-surefire-plugin}</surefire-plugin.version>
+        <compiler-plugin.version>{version.maven-compiler-plugin}</compiler-plugin.version>
```
Comment (maxandersen) [inline devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/maven/base/pom.tpl.qute.xml:19]: why are we doing two different naming strategies here ? 

it should be `maven-compiler.plugin.version` to match what we do everywhere else (except code.quarkus for some reason :)

File: devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/maven/codestart.yml [hunk 3]
```diff
@@ -2,16 +2,19 @@
...
   base:
     data:
-      maven:
-        version:
-          kotlin: 1.3.72
-          scala: 2.12.8
-          scala-maven-plugin: 4.1.1
-          maven-compiler-plugin: 3.8.1
-          maven-surefire-plugin: 2.22.1
+      version:
```
Comment (maxandersen) [inline devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/maven/codestart.yml:11]: that explains the different naming strategy - okey; but then shouldn't the quarkus plugin version be here too ?

**Review by maxandersen — CHANGES_REQUESTED** [review]
looks good; but can we pick one strategy for how we refer to versions please ?

File: devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/maven/codestart.yml (hunk 3 above)
Comment (ia3andy) [reply devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/maven/codestart.yml:11]: @maxandersen the reason behind having `version.name` is that using yaml defining multiple `name.version` isn't really elegant:
```yaml
name1:
  version: 1.0
name2:
  version: 3.0
...
```

The difference is that here we only define versions, for quarkus or platform we also define artifact-id and group-id.

Maybe the `name.version` even if less elegant would make it more consistant?

Comment (ia3andy) [reply devtools/platform-descriptor-json/src/main/resources/bundled-codestarts/buildtool/maven/codestart.yml]: Ok I made the change in a new commit

**Review by maxandersen — APPROVED** [review]
Consistent version fixed this +1

File: integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/CodestartProjectTestRunner.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        }
+
+        public static Wrapper fromBuildtool(String buildtool) {
+            switch (buildtool) {
+                case "maven":
+                    return MAVEN;
+                case "gradle":
+                    return GRADLE;
+                default:
+                    throw new IllegalStateException("No wrapper linked ot that buildtool: " + buildtool);
```
Comment (maxandersen) [inline integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/CodestartProjectTestRunner.java:45]:
```suggestion
                    throw new IllegalStateException("No wrapper linked to buildtool: " + buildtool);
```

**Pull request merged**